docs/source/examples/e2e_flow.rst

```diff
@@ -0,0 +1,519 @@
+    As a result you won't be able to use these quantized models outside of TorchTune.
+    But you should be able to use these with the generation and evaluation recipes within
+    TorchTune. These results will help inform which quantization methods you should use
+    with your favorite inference engine.
+
+Now that we have the quantized model, let's re-run generation.
+
+Modify ``custom_generation_config.yaml`` to include the following changes.
+
+.. code-block:: yaml
+
+    checkpointer:
+        # we need to use the custom TorchTune checkpointer
+        # instead of the HF checkpointer for loading
+        # quantized models
+        _component_: torchtune.utils.FullModelTorchTuneCheckpointer
+
+        # directory with the checkpoint files
+        # this should match the output_dir specified during
+        # finetuning
+        checkpoint_dir: <checkpoint_dir>
+
+        # checkpoint files point to the quantized model
+        checkpoint_files: [
+            hf_model_0001_0-4w.pt,
+        ]
+
+        output_dir: <checkpoint_dir>
+        model_type: LLAMA2
+
+    # we also need to update the quantizer to what was used during
+    # quantization
+    quantizer:
+        _component_: torchtune.utils.quantization.Int4WeightOnlyQuantizer
+        groupsize: 256
+
+
+Once the config is updated, let's kick off generation! We'll use the
+same sampling parameters as before.
+
+We'll use a different prompt from the one in the config
+
+.. code-block:: bash
+
+    tune run generate --config ./custom_generation_config.yaml \
+    prompt="What are some interesting sites to visit in the Bay Area?"
+
+
+Once generation is complete, you'll see the following in the logs.
+
+
+.. code-block:: bash
+
+    [generate.py:92] A park in San Francisco that sits at the top of a big hill.
+                     There are lots of trees and a beautiful view of San Francisco...
+
+    [generate.py:96] Time for inference: 4.13 sec total, 72.62 tokens/sec
+    [generate.py:99] Memory used: 17.85 GB
+
+With quantization (and torch compile under the hood), we've sped up generation
+by almost 3x!
+
+|
+
+Using TorchTune Checkpoints with other libraries
+------------------------------------------------
+
+As we mentioned above, one of the benefits of handling of the checkpoint
+conversion is that users can directly work with standard formats. This helps
+with interoperability with other libraries since TorchTune doesn't add yet
+another format to the mix.
+
+Let's take a look at an example of how this would work with a popular codebase
+used for running performant inference with LLMs -
+`gpt-fast <https://github.com/pytorch-labs/gpt-fast/tree/main>`_. This section
+assumes that you've cloned the repository on your machine.
+
+``gpt-fast`` makes some assumptions about the checkpoint and the availability of
+the key-to-file mapping. Let's satisfy these assumptions, by creating this mapping
```

Comment:
nit: is "key-to-file mapping" defined previously such that it's obvious what "the key-to-file mapping" means? If not just say "a file mapping parameter names to the files containing them" or something like that